builtin/provisioners/chef-client/resource_provisioner.go

```diff
@@ -0,0 +1,402 @@
+package chefclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/hashicorp/terraform/communicator"
+	"github.com/hashicorp/terraform/communicator/remote"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/mitchellh/go-linereader"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	firstBoot      = "first-boot.json"
+	logfileDir     = "logfiles"
```

Comment:
Yes, I think you are missing the actual point/use case for this one. All comments you made to code related to logging, seems to suggest that I'm trying to configure the logging of the Chef Client. But in fact I'm not... If you could have a quick look at the docs in this PR describing the [`log_to_file`](https://github.com/svanharmelen/terraform/blob/f-chef-client-provisioner/website/source/docs/provisioners/chef-client.html.markdown) parameter, that will probably give you a push in the right direction :smiley: 

In short the main purpose of the logging related code in the PR is to enable you to **not** have TF spit out all the Chef Client logging from all nodes on your console at once. If your spinning up 10+ machines which will possibly be build simultaneously in parallel, your console will be on fire with log lines from all nodes running their initial Chef run at the same time. So trying to follow a single host or to scroll back to find out why node x failed, will be undoable.

So if you set the option `log_to_file` to `true` it will redirect those logs to separate files (one per node) and put them in a subdirectory of your current location on your local machine. That way you can review any log of any hosts initial run afterwards if you need to checkup on anything.

So... Long story short, this const is pointing to the location on your local machine where it will create the logs if you enabled it to `log_to_file`. And with that additional info, I would say this is OK as is...